.github/workflows/jdl.yml

```diff
@@ -0,0 +1,247 @@
+jobs:
+  applications:
+    name: ${{ matrix.jdl }}
+    runs-on: ${{ matrix.os }}
+    defaults:
+      run:
+        working-directory: ${{ github.workspace }}/app
+    if: >-
+      !contains(github.event.head_commit.message, '[ci skip]') &&
+      !contains(github.event.head_commit.message, '[skip ci]') &&
+      !contains(github.event.pull_request.title, '[skip ci]') &&
+      !contains(github.event.pull_request.title, '[ci skip]') &&
+      (github.event.pull_request.draft == false || !contains(github.event.pull_request.labels.*.name, 'pr: skip-ci'))
+    timeout-minutes: 40
+    strategy:
+      fail-fast: false
+      matrix:
+        node_version: [14.16.0]
+        os: [ubuntu-20.04]
+        cache: [angular]
+        jdl:
+          - blog-store
+        include:
+          - jdl: blog-store
+            environment: prod
+            # Use monorepository for diff
+            extra-args: --workspaces --monorepository
+            # Backend is failing, disable it
+            skip-backend-tests: 1
+    env:
+      JHI_PROFILE: ${{ matrix.environment }}
+    steps:
+      #----------------------------------------------------------------------
+      # Install all tools and check configuration
+      #----------------------------------------------------------------------
+      - name: 'SETUP: Checkout generator-jhipster'
+        uses: actions/checkout@v2
+        with:
+          path: generator-jhipster
+          fetch-depth: 2
+      - name: 'SETUP: Create required structure'
+        run: |
+          mkdir app
+          mkdir -p base/generator-jhipster
+          mkdir -p base/app
+        working-directory: ${{ github.workspace }}
+      - uses: actions/setup-node@v2.1.5
+        with:
+          node-version: ${{ matrix.node_version }}
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '11.x'
+      - name: 'SETUP: Get date'
+        id: get-date
+        run: |
+          echo "::set-output name=date::$(/bin/date -u "+%Y%m%d")"
+      - name: 'SETUP: load npm cache'
+        uses: actions/cache@v2.1.4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.cache }}-${{ steps.get-date.outputs.date }}-${{ hashFiles('generator-jhipster/package-lock.json', 'generator-jhipster/**/package.json.ejs') }}
+          restore-keys: |
+            ${{ runner.os }}-node-${{ matrix.cache }}-${{ steps.get-date.outputs.date }}-
+      - name: 'SETUP: load maven cache'
+        if: "!contains(matrix.app-type, 'gradle')"
+        uses: actions/cache@v2.1.4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ steps.get-date.outputs.date }}-${{ hashFiles('generator-jhipster/**/pom.xml.ejs') }}
+          restore-keys: |
+            ${{ runner.os }}-maven-${{ steps.get-date.outputs.date }}-
+      - name: 'SETUP: load gradle cache'
+        uses: actions/cache@v2.1.4
```

Comment:
Missing conditional maven/gradle?

Reply:
Improved the logic.

Reply:
I will planning a major change in our caching.